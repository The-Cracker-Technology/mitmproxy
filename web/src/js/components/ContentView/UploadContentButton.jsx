import React from 'react'
import PropTypes from 'prop-types'
import FileChooser from '../common/FileChooser'

UploadContentButton.propTypes = {
    uploadContent: PropTypes.func.isRequired,
}

export default function UploadContentButton({ uploadContent }) {

    return (
        <FileChooser
            icon="fa-upload"
            title="Upload a file to replace the content."
            onOpenFile={uploadContent}
            className="btn btn-default btn-xs"/>
    )
}

