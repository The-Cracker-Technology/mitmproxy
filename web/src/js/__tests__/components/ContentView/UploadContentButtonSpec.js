import React from 'react'
import renderer from 'react-test-renderer'
import UploadContentButton from '../../../components/ContentView/UploadContentButton'

describe('UpdateContentButton Component', () => {
    it('should render correctly', () => {
        let uploadContentFn = jest.fn(),
            uploadContentButton = renderer.create(<UploadContentButton uploadContent={uploadContentFn}/>),
            tree = uploadContentButton.toJSON()
        expect(tree).toMatchSnapshot()
    })
})
