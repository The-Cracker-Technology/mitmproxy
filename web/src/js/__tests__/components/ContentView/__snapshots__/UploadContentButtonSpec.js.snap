// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdateContentButton Component should render correctly 1`] = `
<a
  className="btn btn-default btn-xs"
  href="#"
  onClick={[Function]}
  title="Upload a file to replace the content."
>
  <i
    className="fa fa-fw fa-upload"
  />
  <input
    className="hidden"
    onChange={[Function]}
    type="file"
  />
</a>
`;
